examples/slicing_ndcube.py

```diff
@@ -0,0 +1,128 @@
+# array in indexed) or by real world coordinates. Lets begin by cropping by index.
+
+##############################################################################
+# Cropping by pixel index
+# -------------------
+# Lets say crop the `example_cube` so that we slice out only one wavelength, we can do that
+# by indexing as such
+sliced_cube = example_cube[1, :, :]
+# here we can see we are left with a 2-D cube which is an image at one wavelength.
+sliced_cube.dimensions
+
+# We can also index a region of interest of the cube at a particular wavelength.
+# Again note that we are cropping here based on the `pixel` index rather then than coordinate
+# real world value
+
+sliced_cube = example_cube[1, 10:20, 20:40]
+sliced_cube.dimensions
+
+# now we can inspect the sliced cube, and see its now a smaller region of interest.
+sliced_cube.plot()
+
+##############################################################################
+# Cropping cube using world coordinate values using `ndcube.NDCube.crop()`
+# -------------------
+# Now in many use cases its more useful to crop a cube to a region of interest based
+# on some specific real world coordinates such as coordinates of points in space or over some spectral
+# range. This is achieved by the `ndcube.NDCube.crop()` method which takes high-level astropy coordinate objects,
+# such as an `astropy.coordinate.SkyCoord` object. The `.crop()` returns the smallest cube in pixel space that contains all
+# the passed points.
+
+##############################################################################
+# Lets first define some points over which to crop the `example_cube`. The points are
+# defined as iterables of size two that define a point in the cube.
+# Lets first define two points in space (lat and long) we want to crop but keeping all wavelengths:
+point1 = [SkyCoord(0*u.arcsec, 0*u.arcsec, frame=frames.Helioprojective), None]
+point2 = [SkyCoord(200*u.arcsec, 100*u.arcsec, frame=frames.Helioprojective), None]
+
+cropped_cube = example_cube.crop(point1, point2)
+cropped_cube.dimesions
+
+cropped_cube.plot()
+
+##############################################################################
+# Now lets say we also want to crop over at a  particular wavelength.
+# Lets define a new point, and then include them to be passed in the `.crop` method.
+point3 = [None, SpectralCoord(10.2*u.angstrom)]
+
+cropped_cube = example_cube.crop(point1, point2, point3)
+cropped_cube.dimesions
+
+cropped_cube.plot()
+
+# Now lets say we also want to crop over at a  particular wavelength range.
+# Lets define another new point, and then include them to be passed in the `.crop` method.
+point4 = [None, SpectralCoord(10.6*u.angstrom)]
+
+cropped_cube = example_cube.crop(point1, point2, point3, point4)
+cropped_cube.dimesions
+
+##############################################################################
+# Here we can just see how powerful this can be to easily crop over different world coordinates.
```

Comment:
```suggestion
# Here we can just see how powerful this can be to easily crop over different world coordinates.
# In fact we can make this simpler still by combining the SkyCoords and SpectralCoords into
# just two points:

point5 = [SkyCoord(0*u.arcsec, 0*u.arcsec, frame=frames.Helioprojective), SpectralCoord(10.2*u.angstrom)]
point2 = [SkyCoord(200*u.arcsec, 100*u.arcsec, frame=frames.Helioprojective), SpectralCoord(10.6*u.angstrom)]

cropped_cube = example_cube.crop(point5, point6)
```